compiler/compile/OSRData.hpp

```diff
@@ -44,6 +44,55 @@ namespace TR
       osr,
       traditional
       };
+
+/*
+ * An OSR transition can occur before or after the OSR point,
+ * depending on the current mode.
+ *
+ * In preExecutionOSR, the transition occurs before the OSR point,
+ * resulting in the interpreter resuming execution at the OSR point's
+ * bytecode index and evaluating the instruction it represented.
+ *
+ * In postExecutionOSR, the transition occurs after the OSR point,
+ * resulting in the interpreter resuming execution at the bytecode
+ * index after the OSR point.
+ */
+   enum OSRTransitionTarget
+      {
+      preExecutionOSR,
+      postExecutionOSR
+      };
+
+/*
+ * OSR can operate in two modes, voluntary and involuntary.
+ *
+ * In voluntary OSR, the JIT is aware of the OSR points and controls
+ * the placements of their transitions, limiting the necessary book keeping
+ * to them.
+ *
+ * In involuntary OSR, the transition can occur at any bytecode index, 
+ * requiring the book keeping for a transition at any time.
```

Comment:
Again we need to be clear that involuntary is just that a transition can occur at any potentialOSR point - the generated code does not determine if the transition will take place.